src/marqo/s2_inference/random_utils.py

```diff
@@ -12,8 +12,12 @@ def sentence_to_hash(sentence):
     # for speed reasons we hash
     if isinstance(sentence, ImageType):
         pixel_data = list(sentence.getdata())
-        pixel_averages = [sum(channels)/len(channels) for channels in pixel_data]
-        image_average = functools.reduce(lambda x, y: x + y, pixel_averages)/len(pixel_data)
+        if isinstance(pixel_data[0], tuple):
+            pixel_averages = [sum(channels)/len(channels) for channels in pixel_data]
+            image_average = functools.reduce(lambda x, y: x + y, pixel_averages) / len(pixel_data)
+        else:
+            # uses palette
+            image_average = functools.reduce(lambda x, y: x + y, pixel_data)/len(pixel_data)
```

Comment:
Just clarifying, why are we using `pixel_data` instead of `pixel_averages` here?

Reply:
The data of png file using palette is just a one-dimension array. It does not store the RGB data for each pixel, instead it stores an index of the palette.